Core/clim-core/commands/processor.lisp

```diff
@@ -38,9 +38,14 @@
         (when (and delimiter (delimiter-gesture-p delimiter))
           (read-gesture :stream stream))))
     (with-delimiter-gestures (*command-argument-delimiters* :override t)
-      (setq command-args (funcall (parser (gethash command-name
-                                                   *command-parser-table*))
-                                  stream)))
+      ;; The null context is a barrier which prevents other commands from
+      ;; being selected during reading arguments. -- jd 2020-09-03
+      (with-input-context ('null :override t) ()
+          (setq command-args (funcall (parser (gethash command-name
+                                                       *command-parser-table*))
+                                      stream))
```

Comment:
Will this also prevent commands from being selected as arguments? If not, the comment could be clarified a bit.

Reply:
if an argument type is command, then it will be put as the context by the command parser. In other words barrier cancels all input contexts from above.

Reply:
So maybe "The null context is a barrier which presents other commands from being selected and cancelling the current command while reading arguments. Selecting commands as arguments to the current command still works."?

Reply:
updated the comment to

```
(with-delimiter-gestures (*command-argument-delimiters* :override t)
      ;; The null context is a barrier which prevents other commands from
      ;; being selected during reading arguments and cancelling the current
      ;; command while reading arguments. When the argument's type is command
      ;; then this context will be estabilished by the parser and it will be
      ;; possible to select commands. -- jd 2020-09-03
      (with-input-context ('null :override t) ()
          (setq command-args (funcall (parser (gethash command-name
                                                       *command-parser-table*))
                                      stream))))
```

Reply:
updated again.